src/main/java/everyos.engine.ribbon.renderer.skijarenderer/everyos/engine/ribbon/renderer/skijarenderer/RibbonSkijaRenderer.java

```diff
@@ -119,6 +119,11 @@ private void setClip(int x, int y, int l, int h) {
 		parentRenderer.setClip(x, y, l, h);
 	}
 
+	@Override
+	public void translate(int dx, int dy) {
+		this.x+=dx; this.y+=dy;
```

Comment:
Why can't translate be done via the UI code?
Also, if I'm correct, this won't be clipped properly;.

Reply:
Oh no it actually works just fine!

In overlaying component UI
If I draw anything above y=0, it will be drawn over webicity button!!

That's why I first clip it so that it won't draw over the button.

Later, I can easily draw the round rect above so that it will be clipped, this giving a nice animation 

But this doesn't affect children. I can set their draw positions!!!!!

That's why, when I translate the renderer up, it sets the y=0 over webicity button, but still clipping the part I wanted.

Therefore, children are also drawn nicely.


As such, clipping is working the way I expected it to!